Individual hike view

sambassan opened the pull request:
First card shows hike line and stats, following cards show images with position on line and description. 
Can click navigation arrows on side of image or pagination dots at the bottom to click through images.
Can click on the photo to open a larger modal window of the photo.
Adjusts for mobile/desktop view and light/dark mode.
Pretty beefy view, could probably be componentized a lot better to be more modular, but wanted to get a function view out.

sambassan commented:
Closes #9

apknusel commented:
- fix buttons moving vertically
- use aspect ratio to keep images all the same size
- adjust size of card with viewheight
- add a prop to TraiLine to pass in a lat and long for drawing a point on the line

sambassan commented:
A lot of changes here, a lot of breaking up the previous version into different components on the individual hike page. Major changes:
- Different cards for the different sections of the view.
- Trail Line card remains constant, overlay of picture position updates as you cycle through images, corrected slight offset.
- Fixed image display size/hike stats size so the navigation arrows don't move vertically as you cycle through cards.
- Two columns on desktop sized views to fill more blank space.
- Added composable for formatting hike data from database, refactored PreviousHikesView to use this as well.
- Modified the distance formatters to use km instead of miles.
